src/dmd/expressionsem.d

```diff
@@ -12490,13 +12490,15 @@ bool checkAddressVar(Scope* sc, Expression exp, VarDeclaration v)
                 v.storage_class &= ~STC.maybescope;
                 v.doNotInferScope = true;
                 if (exp.type.hasPointers() && v.storage_class & STC.scope_ &&
+                    !(v.storage_class & STC.temp) &&
```

Comment:
What's this for?

Reply:
Maybe a remnant of the old bug that caused `enclosesLifetimeOf` to return wrong results for `VarDeclaration`s created during semantic?